src/components/tag/tag.js

```diff
@@ -53,80 +58,30 @@ Tag.propTypes = {
-      if (!props.customStyles) {
-        return new Error(
-          `The "customStyles" prop is required when using the "custom" theme in '${componentName}'.`
-        );
-      } else if (typeof props.customStyles !== 'object') {
-        return new Error(
-          `The "customStyles" prop in '${componentName} must be an object'.`
-        );
-      } else if (
-        !props.customStyles.background ||
-        typeof props.customStyles.background !== 'string'
-      ) {
-        return new Error(
-          `The "customStyles.background" prop in '${componentName} is required when using the "custom" theme and must be a string'.`
-        );
-      } else if (
-        !props.customStyles.color ||
-        typeof props.customStyles.color !== 'string'
-      ) {
-        return new Error(
-          `The "customStyles.color" prop in '${componentName} is required when using the "custom" theme and must be a string'.`
-        );
-      } else if (
-        !props.customStyles.borderColor ||
-        typeof props.customStyles.borderColor !== 'string'
-      ) {
-        return new Error(
-          `The "customStyles.borderColor" prop in '${componentName} is required when using the "custom" theme and must be a string'.`
-        );
-      }
-    }
-  },
+  customLabel: (props, currentProp, componentName) =>
+    validateCustomProp(props, currentProp, componentName),
+  /** If the theme is set to "custom", this prop is required. */
+  customTooltipText: (props, currentProp, componentName) =>
+    validateCustomProp(props, currentProp, componentName),
+  /** If the theme is set to "custom", this `customBorder` is required. The value must be a class name. */
+  customBorder: (props, currentProp, componentName) =>
+    validateCustomProp(props, currentProp, componentName),
+  /** If the theme is set to "custom", this `customColor` is required. The value must be a class name. */
+  customColor: (props, currentProp, componentName) =>
+    validateCustomProp(props, currentProp, componentName),
+  /** If the theme is set to "custom", this `customBackground` is required. The value must be a class name. */
+  customBackground: (props, currentProp, componentName) =>
+    validateCustomProp(props, currentProp, componentName),
   /** customIcon is required if using the "custom" theme and `icon=true` */
-  customIcon: (props, componentName) => {
-    if (props.theme === 'custom' && props.icon && !props.customIcon) {
-      return new Error(
-        `The "customIcon" prop is required when using the "custom" theme and "icon" option in '${componentName}'.`
-      );
-    } else if (props.customIcon && typeof props.customIcon !== 'string') {
-      return new Error(
-        `The "customIcon" prop in ${componentName} must be a string'.`
-      );
-    }
-  }
+  customIcon: (props, currentProp, componentName) =>
+    props.icon && validateCustomProp(props, currentProp, componentName)
 };
+
+function validateCustomProp(props, currentProp, componentName) {
```

Comment:
💡 I noticed that custom prop validation were all accomplishing the same thing, so I moved it into a shared function.